File: RegressionMetric.py
from creme import metrics
import pickle
import json

#read configurations
config = json.loads(input())

predict = config['predict']
target = config['target']
m = config['metric']


if (m == "MAE"):
	metric = metrics.MAE()
elif (m == "MSE"):
	metric = metrics.MSE()
elif (m == "RMSE"):
	metric = metrics.RMSE()
elif (m == "RMSLE"):
	metric = metrics.RMSLE()
elif (m == "SMAPE"):
	metric = metrics.SMAPE()


while True:

	#wait request
	data = input()

	Xi = json.loads(data)
	xt = Xi.pop(predict)
	yt = Xi.pop(target)

	print(metric.update(yt, xt))

File: RegressionMetric.js
module.exports = function(RED){
	function RegressionMetricNode(config){
		const path = require('path')
		const utils = require('./utils/utils')

		var node = this;

		//set configurations
		node.file = __dirname + '/RegressionMetric.py'
		node.config = {
			target: config.target || 'y',
			predict: config.predict || 'Predict',
			metric: config.metric || "MAE"
		}
		utils.run(RED, node, config)
	}
	RED.nodes.registerType("RegressionMetric", RegressionMetricNode)
}
